### ISSUE TITLE
Add project signatures

### ISSUE THREAD

**Issue by Jihadik** [supervisely/supervisely#15]
It would be nice to have an ability to understand was dataset changed or not without downloading it. Currently, the easiest solution would be to check if the number of items is the same, however, it's not always precise.

It is nice for models retraining automation.
This answer might be useful: https://stackoverflow.com/questions/545387/linux-compute-a-single-hash-for-a-given-folder-contents

**Comment by antonc-ds** [comment]
Hi @Jihadik . Could you elaborate on your workflow a bit? If you are accessing the data via the API, there is a `updated_at` field in the dataset info that you can look at.
```
>>> api.dataset.get_list(some_project_id)
[DatasetInfo(id=1201, name='ds1', description='', size='861069', project_id=266, images_count=6, created_at='2019-05-29T13:50:10.061Z', updated_at='2019-05-29T13:50:10.061Z')]
```
Same with the whole project, the info tuple contains the modification date.

Or are you dong something completely offline and need to compare datasets strictly by content?

**Comment by Jihadik** [comment]
I didn’t notice `updated_at` field. That should be sufficiently good!

Our workflow is the following:
1. Monitor changes in a specific workspace/project via your API where segmented/classified datasets are uploaded
2. Once a change is detected, download it and start a training process
3. Once the training is finished, validate the final model and deploy it to production if it became better

We could train everything through your API via the agent, but rewriting everything to your format is painful and packing it into docket images and validating that everything is working correctly would require a lot of time.

We already have our working training process, so the workflow I described was the easiest.

P.S. The issue is resolved, feel free to close it.

**Comment by antonc-ds** [comment]
Sounds good, thanks for the details!
